lib/global_id/uri/gid.rb

```diff
@@ -83,7 +83,7 @@ def create(app, model, params = nil)
       def build(args)
         parts = Util.make_components_hash(self, args)
         parts[:host] = parts[:app]
-        parts[:path] = "/#{parts[:model_name]}/#{parts[:model_id]}"
+        parts[:path] = "/#{parts[:model_name]}/#{CGI.escape(parts[:model_id].to_s)}"
```

Comment:
Should we do `:model_name` while we're here, for completeness?

If it's a ruby constant name, it will obviously be URL-safe already... but in principle we could be building a GID reference to some external app written in some other language.. I think?